src/TropicalGeometry/semiring.jl

```diff
@@ -0,0 +1,504 @@
+  end
+  return Expr(:call, "", expressify(data(x), context = context))
+end
+
+AbstractAlgebra.expressify(R::TropicalSemiring{typeof(min)}; context = nothing) = "Tropical ring (min)"
+
+AbstractAlgebra.expressify(R::TropicalSemiring{typeof(max)}; context = nothing) = "Tropical ring (max)"
+
+@enable_all_show_via_expressify TropicalSemiringElem
+
+@enable_all_show_via_expressify TropicalSemiring
+
+
+################################################################################
+#
+#  Predicates
+#
+################################################################################
+
+Oscar.iszero(x::TropicalSemiringElem) = isinf(x)
+
+Oscar.isone(x::TropicalSemiringElem) = !isinf(x) && iszero(data(x))
+
+################################################################################
+#
+#  Equality
+#
+################################################################################
+
+function Base.:(==)(x::TropicalSemiringElem, y::TropicalSemiringElem)
+  (isinf(x) && isinf(y)) && return true
+  ((isinf(x) && !isinf(y)) || (!isinf(x) && isinf(y))) && return false
+  return data(x) == data(y)
+end
+
+################################################################################
+#
+#  Comparison
+#    * min is ordered as usual: -inf < ... < -1 < 0 < 1 < ...
+#    * max is ordered reversely:       ... > -1 > 0 > 1 > ... > inf
+#    (see Section 2.7 in Joswig: "Essentials of Tropical Combinatorics"
+#
+################################################################################
+
+function isless(x::TropicalSemiringElem{typeof(min)}, y::TropicalSemiringElem{typeof(min)})
+  if isinf(x)
+    return false # x=-inf, no y is smaller
+  end
+  if isinf(y)
+    return !isinf(x) # y=-inf, smaller than all x except x=-inf
+  end
+  return data(x) < data(y)
+end
+
+function isless(x::TropicalSemiringElem{typeof(max)}, y::TropicalSemiringElem{typeof(max)})
+  if isinf(x)
+    return false # x=inf, no y is smaller
+  end
+  if isinf(y)
+    return !isinf(x) # y=inf, smaller than all x except x=inf
+  end
+  return data(x) > data(y)
+end
+
+################################################################################
+#
+#  Copying
+#
+################################################################################
+
+Base.copy(a::TropicalSemiringElem) = a
+
+function Base.deepcopy_internal(x::TropicalSemiringElem, dict::IdDict)
+  if !isinf(x)
+    return TropicalSemiringElem(x.parent, Base.deepcopy_internal(data(x), dict))
+  else
+    return inf(parent(x))
+  end
+end
+
+################################################################################
+#
+#  Addition
+#
+################################################################################
+
+function Base.:(+)(x::TropicalSemiringElem{T}, y::TropicalSemiringElem{T}) where {T}
+  if isinf(x)
+    return deepcopy(y)
+  else
+    if isinf(y)
+      return deepcopy(x)
+    else
+      return parent(x)(convention(parent(x))(data(x), data(y)))
+    end
+  end
```

Comment:
Style remark: I would recommend unnecessary nesting. So:
```suggestion
  if isinf(x)
    return deepcopy(y)
  elseif isinf(y)
    return deepcopy(x)
  else
    return parent(x)(convention(parent(x))(data(x), data(y)))
  end
```
or even
```suggestion
  isinf(x) && return deepcopy(y)
  isinf(y) && return deepcopy(x)
  return parent(x)(convention(parent(x))(data(x), data(y)))
```